### PR TITLE
Match actor_api.c

### PR DESCRIPTION
**Pull request by JoshDuMan**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 14 functions (+0.10%) on `us`.

✅ This PR fixes 12 warnings!

File: src/actor_api.c [hunk 1]
```diff
@@ -2249,16 +2488,47 @@ ApiStatus GetTargetListLength(Evt* script) {
     return ApiStatus_DONE2;
 }
 
-INCLUDE_ASM(s32, "197F40", GetOwnerTarget);
+ApiStatus GetOwnerTarget(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    Actor* temp_s2 = get_actor(script->owner1.enemyID);
+    s32 temp = *args++;
+    s32 temp2 = *args++;
```
Comment (nanaian) [inline src/actor_api.c:2495]: Are these required? If not can you name them & `temp_s2`

File: src/actor_api.c [hunk 2]
```diff
@@ -2321,9 +2608,43 @@ ApiStatus func_8026EBF8(Evt* script, s32 isInitialCall) {
...
+    }
+
+    actorPart = get_actor_part(get_actor(actorID), temp_s0_3);
+    if (temp_s3) {
+        actorPart->flags |= ACTOR_FLAG_1000000;
+    } else {
+        actorPart->flags &= ~ACTOR_FLAG_1000000;
+    }
+
+
```
Comment (nanaian) [inline src/actor_api.c:2630]: Remove this newline

**Review by nanaian — CHANGES_REQUESTED** [review]
0.1%, gg.

Please also name this file src/**evt**/actor_api.c

File: src/actor_api.c [hunk 3]
```diff
@@ -2238,9 +2324,162 @@ ApiStatus SetOwnerTarget(Evt* script, s32 isInitialCall) {
...
+        phi_v1_4 = actor->targetIndexList[0];
+        target = &actor->targetData[phi_v1_4];
+        actor->targetActorID = target->actorID;
+        actor->targetPartIndex = target->partID;
+        return ApiStatus_DONE2;
+    }
+
+    if (temp_v0 == 0xA) {
+        actor->selectedTargetIndex = actor->targetListLength - 1;
+        phi_v1_4 = actor->targetIndexList[ actor->selectedTargetIndex];
```
Comment (ethteck) [inline src/actor_api.c:2349]:
```suggestion
        phi_v1_4 = actor->targetIndexList[actor->selectedTargetIndex];
```

File: src/actor_api.c [hunk 4]
```diff
@@ -2238,9 +2324,162 @@ ApiStatus SetOwnerTarget(Evt* script, s32 isInitialCall) {
...
+        phi_a2 = -1;
+    }
+
+    if (phi_v1 >= temp_v0_3) {
+        phi_v1 = 0;
+        phi_a2 = -1;
+    }
+
+    actor->selectedTargetIndex = phi_v1;
+    target = &actor->targetData[actor->targetIndexList[(s8) phi_v1]];
```
Comment (ethteck) [inline src/actor_api.c:2377]: is this cast needed? maybe the type should be different?

**Review by ethteck — CHANGES_REQUESTED** [review]
Great work here! The only other thing I'd say is that I bet some of the temps in ChooseNextTarget are unnecessary. but no worries if you can't remove any of them.

**Pull request merged**